test/helpers/mock_server.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'webrick'
+
+module MockServer
```

Comment:
My only issue here is poor encapsulation. `cleanup` technically expects a thread id for something started by `MockServer`, but really we could give it any thread id and things will be fine.

A much better design is to return instances of `MockServer`. You can simplify construction through a class method, but hide the thread id in the class:

```ruby
class MockServer
  class << self
    def start(hostname:, port:)
      new.tap do |server|
        server.start
        server.poll_until_ready
        puts "Created test server #{tid}, port: #{port}"
      end
    end
  end

  def initialize(hostname:, port:)
    @hostname = hostname
    @port = port
    @tid = nil
  end

  def start
    @tid = Thread.new do
      start_server(hostname, port)
    end
  end

  def stop
    Thread.kill(@tid)
  end

  ...
end
```